core/src/main/java/com/spotify/ffwd/output/CoreOutputManager.java

```diff
@@ -220,6 +292,22 @@ private boolean rateLimitAllowed(int permits, String key) {
         }
     }
 
+    private boolean cardinalityLimitAllowed(long currentCardinality) {
+        return cardinalityLimit == null || cardinalityLimit >= currentCardinality;
+    }
+
+    /**
+    * To reset cardinality this will swap HLL++ if it was tripped after configured period of ms
+    */
+    private void swapHyperLogLogPlus() {
```

Comment:
@hexedpackets It looks ugly and I'm curious if there is other way to atomically swap the objects

Reply:
hmm, I can't think of a cleaner way to do it

Reply:
👍